src/do_turn.cpp

```diff
@@ -1007,10 +1007,16 @@ bool do_turn()
         if( u.activity.is_interruptible() && u.activity.interruptable_with_kb ) {
             wait_message += string_format( _( "\n%s to interrupt" ), press_x( ACTION_PAUSE ) );
         }
-        wait_refresh_rate = 5_minutes;
+        static const activity_id autodrive_activity_id( "ACT_AUTODRIVE" );
+        if( u.activity.id() == autodrive_activity_id ) {
+            wait_refresh_rate = 1_turns;
+        } else {
+            wait_refresh_rate = 5_minutes;
+        }
     }
     if( wait_redraw ) {
-        if( g->first_redraw_since_waiting_started || calendar::once_every( 1_minutes ) ) {
+        if( g->first_redraw_since_waiting_started || wait_refresh_rate <  1_minutes ||
+            calendar::once_every( 1_minutes ) ) {
```

Comment:
```suggestion
        if( g->first_redraw_since_waiting_started || calendar::once_every( std::min( 1_minutes, wait_refresh_rate ) ) ) {
```
Currently has the same effect, but this might be better if `wait_refresh_rate` is set to multiple turns so that the popup is not redrawn too many times during a few turns.

Reply:
Done.